### DISCUSSION TITLE
View/Edit/Clone

### DISCUSSION THREAD

**Discussion by PaulMcF1987** [Budibase/budibase#5977] [Help]
I have asked a similar question before but am still struggling to find a solution.

The app I am creating is to keep a log of all of my customers and the jobs that have been carried out.

I then need 3 "views"

- view - to view the job details
- edit - to edit/ammend details
- clone - to duplicate the job if another customer has a similar need or if the same job comes up again.

Currently when I click to view an entry, I have 3 forms displayed on the screen... one for each of the above

Ideally I would like to only have the "view" form displayed and on this form, 2 buttons "edit" and "clone"

I have tried using steps, but since the steps need to be wrapped in a form, which can either be set to "update" or "Create" it seems as though I cannot have both the edit ad clone options, I need one or the other???

The next thing I was considering was setting the CSS of "edit" and "clone" to `display:none;`

Then when I press one of the buttons it should set the "view" form to `display:none` and either the edit or clone for to `display:contents`

However, I cant find a way to avhieve this as It doesnt seem to allow for custom JS to be inserted into a button

Does anyone know how to make the above work?

Or a workaround that minght achieve the same resullt?

**Comment by melohagan** [comment]
Firstly I wouldn't worry about the 'Create/Update' type. It isn't as important as it seems - I think it's more of a backend nuance.
I raised an issue here: https://github.com/Budibase/budibase/issues/5521 but I do need to follow up on it.

You can show/hide forms, containers, and buttons using conditional UI: https://docs.budibase.com/docs/conditions

**Reply by PaulMcF1987** [reply]
I had a look at the conditional UI but I cant find a way to link the conditional UI to perform on a button click.
It seems to operate more based on the conditions set by the logged in user or the information which is on the form.

I have tried adding a hidden field that changes value on click and set the terms to respond to the value but no joy. the form seems to stay hidden, I cant find a handlebar option to link to cell to the button click un the conditional UI

Based on what you said above, should the form steps work for this?

**Reply by melohagan** [reply]
To get conditional UI to work, you could set <a href="https://docs.budibase.com/docs/app-state">App State</a> on your button click, and have the condition based on that state binding. 

Form Steps should also work as well. https://docs.budibase.com/docs/form-steps
